### PR TITLE
[final][hold] subscriber attributes

### PR DESCRIPTION
**Pull request by aboedo**
Adds subscriber attributes and `invalidatePurchaserInfoCache` to the react native plugin. 
Updates the versions of `purchases-hybrid-common` to the latest. 

tested on device and it's working correctly

### Requirements

- [x] [~purchases-hybrid-common 1.0.7~](https://github.com/RevenueCat/purchases-hybrid-common/pull/5)
- [x] ~purchases-ios 3.1.2~
- [x] `purchases-hybrid-common` PR that handles `setAttributes` being called with `NSNull` values

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -5,11 +5,11 @@ if (os.type() === "Linux") {
   console.log("Skipping iOS Dependencies");
 } else if (os.type() === "Darwin") {
   const downloadProcess = exec(
-    "./scripts/download-purchases-framework.sh 3.0.1"
+    "./scripts/download-purchases-framework.sh 3.1.1"
```
Comment (aboedo) [inline build.js]: gonna update this soon to 3.1.2 so that it includes the `setPushTokenString` method from 3.1.2

File: build.js [hunk 2]
```diff
@@ -5,11 +5,11 @@ if (os.type() === "Linux") {
...
   );
   downloadProcess.stdout.pipe(process.stdout);
   const downloadProcessCommon = exec(
-    "./scripts/download-purchases-common.sh 1.0.5"
+    "./scripts/download-purchases-common.sh dev_subscriber_attributes"
```
Comment (aboedo) [inline build.js]: will update the pointer here when a release is out

File: android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java [hunk 3]
```diff
@@ -202,6 +203,46 @@ public void onReceived(@NonNull PurchaserInfo purchaserInfo) {
                 .emit(RNPurchasesModule.PURCHASER_INFO_UPDATED, convertMapToWriteableMap(MappersKt.map((purchaserInfo))));
     }
 
+    @ReactMethod
+    public void invalidatePurchaserInfoCache() {
+        CommonKt.invalidatePurchaserInfoCache();
+    }
+
```
Comment (vegaro) [inline android/src/main/java/com/revenuecat/purchases/react/RNPurchasesModule.java]: minor: extra space

File: package.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react-native-purchases",
   "title": "React Native Purchases",
-  "version": "3.0.6",
+  "version": "3.0.7",
```
Comment (vegaro) [inline package.json]: Should we make it 3.1.0, since we made minor updates in iOS and Android?

File: scripts/download-purchases-framework.sh [hunk 5]
```diff
@@ -28,7 +28,7 @@ curl -sSL $URL > temp.zip
 # In some cases the temp folder can not be created by unzip, https://github.com/RevenueCat/react-native-purchases/issues/26
 mkdir -p temp
 unzip -o temp.zip -d temp
-mv temp/Purchases.framework ./Purchases.framework
+mv temp/Carthage/Build/iOS/Purchases.framework ./Purchases.framework
```
Comment (vegaro) [inline scripts/download-purchases-framework.sh]: was this broken? omg

File: src/index.ts [hunk 6]
```diff
@@ -840,4 +840,65 @@ export default class Purchases {
...
+  public static setAttributes(attributes: { [key: string]: string }) {
+    RNPurchases.setAttributes(attributes);
+  }
+
+  /**
+   * Subscriber attribute associated with the email address for the user
+   *
+   * @param email Empty String or null will delete the subscriber attribute.
+   */
+  public static setEmail(email: string) {
```
Comment (vegaro) [inline src/index.ts]: I would indicate that attributes can also be null as `string | null`

**Comment by vegaro** [comment]
I would add some tests to https://github.com/RevenueCat/react-native-purchases/blob/master/__tests__/index.test.js to make sure de right calls are being made.

Mock the new calls here https://github.com/RevenueCat/react-native-purchases/blob/master/setupJest.js#L673 and make sure you verify they are being called.

**Comment by robertveloso** [comment]
Any date for publish?

**Comment by vegaro** [comment]
@robertveloso No specific date, but this week

File: src/index.ts (hunk 6 above)
Comment (aboedo) [reply src/index.ts]: awesome! didn't know you could do that. will do

File: scripts/download-purchases-framework.sh (hunk 5 above)
Comment (aboedo) [reply scripts/download-purchases-framework.sh]: yeah, but the breaking was fairly recent - it came from the iOS release automation stuff

File: VERSIONS.md [hunk 7]
```diff
@@ -14,4 +14,4 @@
 | 2.3.3   | 2.5.0       | 2.3.1           | 0.1.2                |
 | 2.3.2   | 2.5.0       | 2.3.0           | 0.1.2                |
 | 2.3.1   | 2.5.0       | 2.3.0           | 0.1.1                |
-| 2.3.0   | 2.4.0       | 2.3.0           | N/A                  |
\ No newline at end of file
+| 2.3.0   | 2.4.0       | 2.3.0           | N/A                  |
```
Comment (vegaro) [inline VERSIONS.md]: Why is this a change lol

File: android/src/main/java/com/revenuecat/purchases/common/mappers.kt [hunk 8]
```diff
@@ -10,6 +10,7 @@ import com.revenuecat.purchases.PurchaserInfo
 import com.revenuecat.purchases.util.Iso8601Utils
 import java.text.NumberFormat
 import java.util.Currency
+import org.json.JSONObject
```
Comment (vegaro) [inline android/src/main/java/com/revenuecat/purchases/common/mappers.kt]: Did you copy over the updated common files?

File: VERSIONS.md (hunk 7 above)
Comment (aboedo) [reply VERSIONS.md]: some weird leftover from when I had actually changed the version number. I'll roll it back. My editor is set to always add an empty line at EoF

**Pull request merged**